src/v/raft/state_machine.cc

```diff
@@ -128,7 +128,7 @@ ss::future<> state_machine::apply() {
           return f.then([this] {
               // build a reader for log range [_next, +inf).
               storage::log_reader_config config(
-                _next, model::model_limits<model::offset>::max(), _io_prio);
+                _next, _raft->committed_offset(), _io_prio);
```

Comment:
nit: comment above needs updation... probably good to add a small comment on why committed_offset .

Reply:
added